packages/zoe/src/z2spec/escrow-exo.js

```diff
@@ -0,0 +1,311 @@
+/**
+ * @file escrow exchange using ownable seat, resolver
+ */
+// @ts-check
+
+import { AmountMath } from '@agoric/ertp';
+import { deeplyFulfilledObject } from '@agoric/internal';
+import { prepareDurablePublishKit } from '@agoric/notifier';
+import { assert, bare, Fail, q } from '@endo/errors';
+import { E } from '@endo/far';
+import { M, objectMap } from '@endo/patterns';
+import {
+  addToAllocation,
+  subtractFromAllocation,
+} from '../contractFacet/allocationMath.js';
+import { isOfferSafe } from '../contractFacet/offerSafety.js';
+import { prepareOwnable } from '../contractSupport/prepare-ownable.js';
+
+/**
+ * @import {Zone} from '@agoric/base-zone'
+ */
+
+const { keys, entries, fromEntries } = Object;
+const { isEmpty } = AmountMath;
+
+const objectFilter = (obj, pred) =>
+  fromEntries(entries(obj).filter(([k, v]) => pred(v, k)));
+
+/**
+ * @param {Zone} zone
+ * @param {<T>() => PublishKit<T>} makeDurablePublishKit
+ */
+export const prepareSeatKit = (zone, makeDurablePublishKit) =>
+  zone.exoClassKit(
+    'Seat Kit',
+    undefined,
+    proposal => {
+      /** @type {PublishKit<unknown>} */
+      const exitKit = makeDurablePublishKit();
+      /** @type {PublishKit<PaymentPKeywordRecord>} */
+      const payoutsKit = makeDurablePublishKit();
+      return {
+        proposal,
+        exitKit,
+        payoutsKit,
+        exiting: false,
+        /** @type {Allocation} */
+        currentAllocation: proposal.give,
+      };
+    },
+    {
+      readOnly: {
+        hasExited() {
+          return this.state.exiting;
+        },
+        getProposal() {
+          return this.state.proposal;
+        },
+        getCurrentAllocation() {
+          return this.state.currentAllocation;
+        },
+        getExitSubscriber() {
+          return this.state.exitKit.subscriber;
+        },
+      },
+
+      /** The right to collect payouts */
+      seat: {
+        /** @param {Keyword} kw */
+        async getPayout(kw) {
+          const { payoutsKit } = this.state;
+          const paid = await payoutsKit.subscriber.getUpdateSince();
+          return paid.value[kw];
+        },
+        async getPayouts() {
```

Comment:
`getPayout` and `getPayouts` are unfortunately not compat with exclusive transferability of the right to collect payouts, unless we do something weird to revoke these atomically on ownership transfer. Since an exited seat is not transferable (which makes sense since it represents no remaining rights), we could somehow ensure that all payout promises dispensed before a transfer (and therefore before exit) do not become connected to the payouts that appear at exit. But this would need more ownables mechanism. I'm not sure what. Interesting!